internal/ingress/controller/parser/parser.go

```diff
@@ -1682,26 +1715,46 @@ func toKongPlugin(plugin plugin) kong.Plugin {
 	return result
 }
 
-func kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) kong.Plugin {
+func (p *Parser) kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) (kong.Plugin, error) {
+	var err error
+	if k8sPlugin.ConfigFrom != (configurationv1.NamespacedSecretValueFromSource{}) {
+		if len(k8sPlugin.Config) > 0 {
+			err = errors.Errorf("plugin '%v' has both Config and ConfigFrom set", k8sPlugin.Name)
```

Comment:
This control flow is not idiomatic for `go`.
Either return at this point in this method or log it and continue.

Meta-question:
What is the behavior when both are set? Are we not configuring that error or is there a precedence?